src/System.Linq/src/System/Linq/ElementAt.cs

```diff
@@ -64,7 +35,8 @@ public static TSource ElementAtOrDefault<TSource>(this IEnumerable<TSource> sour
             if (partition != null)
             {
                 bool found;
-                return partition.TryGetElementAt(index, out found);
+                element = partition.TryGetElementAt(index, out found);
```

Comment:
@JonHanna After this PR is merged, would it make sense to change `IPartition` to follow a more conventional pattern of returning a bool and out-ing a `TSource`, and change all of its names from `TryGet*` to `Try*`?